account/account_test.go

```diff
@@ -579,3 +580,40 @@ func TestAccountUpgradeStatusGetAccount(t *testing.T) {
 		})
 	}
 }
+
+func TestDeprecateEventsEnabledField(t *testing.T) {
+
+	testCases := []struct {
+		name    string
+		account *config.Account
+		want    *bool
+	}{
+		{
+			name: "events.enabled is nil",
+			account: &config.Account{
+				EventsEnabled: ptrutil.ToPtr(true),
+				Events: config.Events{
+					Enabled: nil,
+				},
+			},
+			want: ptrutil.ToPtr(true),
+		},
+		{
+			name: "events.enabled is non-nil",
+			account: &config.Account{
+				Events: config.Events{
+					Enabled: ptrutil.ToPtr(true),
+				},
+				EventsEnabled: ptrutil.ToPtr(false),
+			},
+			want: ptrutil.ToPtr(true),
+		},
```

Comment:
As is, `TestDeprecateEventsEnabledField` covers the following scenarios:

| account | EventsEnabled | Events.Enabled |  Output  |
| ------- | ------------- | -------------- | -------- | 
| not-nil |     true      |     nil        |   true   |
| not-nil |     false     |     true       |   true   |

Can we expand its coverage to the full set of possibilities shown below?

| account | EventsEnabled           | Events.Enabled in       | Events.Enabled out      |
| ------- | ----------------------- | ----------------------- | ----------------------- | 
|  nil    |     N/A                 |     N/A                 |   nil                   |
| not-nil |     nil                 |     nil                 |   nil                   |
| not-nil |     nil                 | not-nil (true or false) | Events.Enabled in |
| not-nil | not-nil (true or false) |     nil                 | EventsEnabled value     |
| not-nil | not-nil (true or false) | not-nil (true or false) | vents.Enabled in  |